tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -240,30 +243,25 @@ public async Task LoadModelAsync(bool useDebugValues = false)
             await RunSyncTimmer();
         });
 
-        foreach (var providerViewModel in PerProviderViewModels)
+        lock (ComputeSystemCards)
         {
-            var computeSystemCards = providerViewModel.ComputeSystems;
-            lock (computeSystemCards)
+            for (var i = ComputeSystemCards.Count - 1; i >= 0; i--)
             {
-                for (var i = computeSystemCards.Count - 1; i >= 0; i--)
+                if (ComputeSystemCards[i] is ComputeSystemViewModel computeSystemViewModel)

```

Comment:
NIT: ComputeSystemCards.Where(x => x in ComputerSystemViewModel).Foreach(x => x.RemoveStateChangedHandler(); x.ComputeSystemErrorReceived -= OnComputeSystemOperationError;).Remove();

Maybe.  I don't know if .Remove can be appended to this list.  I don't have the code up to test.